tgui/packages/tgui/interfaces/RapidTilingDevice.tsx

```diff
@@ -0,0 +1,181 @@
+import { classes } from 'common/react';
+import { capitalizeAll } from 'common/string';
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Section, Tabs, Stack } from '../components';
+import { InfoSection } from './RapidConstructionDevice';
+import { Window } from '../layouts';
+
+type Data = {
+  selected_icon: string;
+  tile_dirs: string[];
+  selected_direction: string;
+  selected_category: string;
+  selected_recipe: string;
+  categories: Category[];
+};
+
+type Category = {
+  category_name: string;
+  recipes: Design[];
+};
+
+type Design = {
+  name: string;
+  icon: string;
+};
+
+const TilePreview = (props, context) => {
+  const { data } = useBackend<Data>(context);
+  const { selected_icon, selected_direction } = data;
+  const rotation = (direction) => {
+    if (direction === 'north') {
+      return 'rotateZ(-180deg)';
+    } else if (direction === 'west') {
+      return 'rotateZ(90deg)';
+    } else if (direction === 'east') {
+      return 'rotateZ(-90deg)';
+    } else {
+      return '';
+    }
+  };
```

Comment:
Make it a global?
```const ROTATION_MAP = {
    north: 'rotateZ(-180deg)';
    west: 'rotateZ(90deg)';
    east: 'rotateZ(-90deg)';
} as const;
```

then inside TilePreview
`const rotation = ROTATION_MAP[direction] || '';`

Reply:
You sure north doesn't need to be wrapped in single quotes I.e 'north'? direction is of type string

Reply:
No point making it global when it's only used inside this component

Reply:
Getting real late.  I'll do this tom

Reply:
> No point making it global when it's only used inside this component

either way is fine, truly. but this sort of switch/if else chain to make a mapping of strings can be made cleaner

also, not sure why north is highlighted here, it should be ok to use without single quotes.

Reply:
No reason to highlight just wanted to give an example. 
No single quote it is then 
I'll draft this for now so I can properly finish this tom

Reply:
it's a bug with using semicolons. github thinks you're doing a new statement

```ts
const ROTATION_MAP = {
    north: 'rotateZ(-180deg)',
    west: 'rotateZ(90deg)',
    east: 'rotateZ(-90deg)',
} as const;
```

Reply:
oh oops! that would do it